### PR TITLE
[Feature/pur 610 textarea]: textareaV2 component

### PR DESCRIPTION
**Pull request by kevin-ho87**
TextareaV2 component.
Contains animating label, error message

image

image

image

### CODE REVIEW

File: src/components/TextareaV2/TextareaV2.fixture.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+const TextareaV2Fixture = {};
```
Comment (MkMan) [inline src/components/TextareaV2/TextareaV2.fixture.ts:1]: Is this file needed if it's empty?

Comment (MkMan) [inline src/components/TextareaV2/TextareaV2.fixture.ts:1]: Is this file needed if it's empty?

File: src/components/TextareaV2/TextareaV2.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+      onChange,
+      rows,
+      theme = 'dark',
+      value: valueProp,
+    },
+    ref,
+  ) => {
+    const uid = useUID();
+    const textareaId = idProp || uid;
+    const [value, setValue] = useState(valueProp || '');
```
Comment (MkMan) [inline src/components/TextareaV2/TextareaV2.tsx:28]: Would using nullish coalescing be better? Otherwise a falsy value of `0` would end up being `''`

```suggestion
    const [value, setValue] = useState(valueProp ?? '');
```

**Review by MkMan — COMMENTED** [review]
This is some fine work 👌 

Just one minor thing, when I ran it locally I couldn't notice a difference in styling when the component was enabled vs disabled

Textarea disabled

File: src/components/TextareaV2/TextareaV2.fixture.ts (hunk 1 above)
Comment (kevin-ho87) [reply src/components/TextareaV2/TextareaV2.fixture.ts:1]: no its not needed. i will delete

**Comment by cyrusxzw** [comment]
Screen Shot 2021-11-30 at 9 46 27 am

I run at my end, and text is cropped a bit at the top in mobile view.

**Comment by cyrusxzw** [comment]
Nice work 👍. Just wonder how to fix 'Similar blocks of code found in 2 locations' in codeclimate. I have the same issue in my PR.

**Comment by kevin-ho87** [comment]
> Nice work 👍. Just wonder how to fix 'Similar blocks of code found in 2 locations' in codeclimate. I have the same issue in my PR.

Thanks, yeah its tricky. I am kinda testing different things

**Comment by MkMan** [comment]
> Nice work 👍. Just wonder how to fix 'Similar blocks of code found in 2 locations' in codeclimate. I have the same issue in my PR.

I'd only worry about it if it's highlighting similar code blocks in the same file, otherwise don't stress 👍 

FYI @kevin-ho87 @cyrusxzw

**Review by MkMan — APPROVED** [review]
